### DISCUSSION TITLE
location path is not working

### DISCUSSION THREAD

**Discussion by eluft84** [nginx/nginx#286] [General]
### Environment

Include the result of the following commands:
  - `nginx -V` 1.26

### Description

I´m trying to setup nginx without success :disappointed:. I have two services behind my proxy: HA and Wiki . HA works fine with IP/ but IP/wiki is not working at all:

- [X ] The bug is reproducible with the latest version of nginx

#### nginx configuration

```
http {

  server {
    listen 80;
    server_name docker.compose.local;
    return 301 https://$host$request_uri;
  }

  server {
    listen 443 ssl;
    server_name docker.compose.local;

    ssl_certificate /etc/nginx/certs/SOME.CRT;
    ssl_certificate_key /etc/nginx/certs/SOME.KEY;
    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
    ssl_ciphers HIGH:!aNULL:!MD5;

    location / {
      proxy_buffering off;
      proxy_set_header X-Forwarded-Proto $scheme;
      proxy_set_header X-Forwarded-Host $host;
      proxy_set_header X-Forwarded-Port $server_port;

      proxy_pass http://192.168.178.184:8123/;
    }

    location /wiki {
      proxy_buffering off;
      proxy_set_header X-Forwarded-Proto $scheme;
      proxy_set_header X-Forwarded-Host $host;
      proxy_set_header X-Forwarded-Port $server_port;

      include /etc/nginx/mime.types;

      proxy_pass http://192.168.178.185/;

    }
}
}
```

If i try to open IP/wiki I see a lot of mime type issues:
```
refused to apply style from 'https://LINK/_assets/css/app.812159e8900d2514eb6d.css' because its MIME type ('text/plain') is not a supported stylesheet MIME type, and strict MIME checking is enabled.Understand this error
runtime.js:1 
        
        
       Failed to load resource: the server responded with a status of 404 ()Understand this error
wiki/:1 Refused to execute script from 'https://LINK/_assets/js/runtime.js?1728723472' because its MIME type ('text/plain') is not executable, and strict MIME type checking is enabled.Understand this error
app.js:1 
        
        
       Failed to load resource: the server responded with a status of 404 ()Understand this error
wiki/:1 Refused to execute script from 'https://LINK/_assets/js/app.js?1728723472' because its MIME type ('text/plain') is not executable, and strict MIME type checking is enabled.Understand this error
manifest.json:1 
        
        
       Failed to load resource: the server responded with a status of 404 ()Understand this error
manifest.json:1 Manifest: Line: 1, column: 4, Unexpected data after root element.Understand this error
wiki/:1 Unchecked runtime.lastError: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was receivedUnderstand this error
wiki/:1 Unchecked runtime.lastError: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was receivedUnderstand this error
wiki/:1 Refused to apply style from 'https://home.luft.cc/_assets/css/app.812159e8900d2514eb6d.css' because its MIME type ('text/plain') is not a supported stylesheet MIME type, and strict MIME checking is enabled.Understand this error
manifest.json:1 
        
        
       Failed to load resource: the server responded with a status of 404 ()Understand this error
```
If i call http://192.168.178.185/ from my host it works. (wiki.js)

**Comment by viktorberezikov** [comment]
This issue is not an nginx issue. This is a configuration issue and a discussion section would be more appropriate for a question like this. Therefore, please ask new questions there.

You access your backend `192.168.178.185` through the links `/wiki`. The links your backend returns for your browser do not contain the `/wiki` part and go to another location `/` with the backend `192.168.178.184:8123`. It seems that your second backend doesn't expect it and returns the 404 code. I don't know why you need a second backend, so I'll just suggest testing the following example with only one location:
```
server { 
    location / {
        proxy_pass http://192.168.178.185;
        # ...
    }
}
```
Just in case, I guess you don't need a slash at the end of the `proxy_pass` directive.
More information here:
https://nginx.org/r/location

**Comment by eluft84** [comment]
If I route / to http://192.168.178.185:3000 it works.

My challenge: I have one domain and nginx is listening to it. My goal is to use this domain and different location paths to forward to right service:

DOMAIN/ --> 192.168.178.184:8123
DOMAIN/WIKI --> 192.168.178.185:3000
DOMAIN/OTHERS --> 192.168.178.180:5000

**Comment by eluft84** [comment]
still not working and didn´t find any settings I have to set on the corresponding service. 
I´m confused - is that not one of the use-cases of a proxy server?

**Comment by viktorberezikov** [comment]
You configure locations in nginx to access different backends. And I'm sure it's working. The issue is that your backend returns pages that do not contain the appropriate prefix (`/wiki/` or `/others/`) to the links. And the client/browser makes a request to nginx, which goes to another location and backend (for example, just `/`). Accordingly, your backend should add a prefix to all links for all pages.
If your backends cannot add the prefix you have configured in nginx, then I would suggest using virtual servers in nginx for different backends.

**Reply by eluft84** [reply]
Not working :-( 
Can you give me a little more information or example of "then I would suggest using virtual servers in nginx for different backends"?

**Reply by viktorberezikov** [reply]
You can read more about virtual servers here: https://nginx.org/en/docs/http/request_processing.html
